packages/framework/src/Model/Product/ProductFacade.php

```diff
@@ -354,9 +354,26 @@ public function getAllProductSellingPricesIndexedByDomainId(Product $product)
         $productSellingPrices = [];
 
         foreach ($this->pricingGroupRepository->getAll() as $pricingGroup) {
-            $productSellingPrices[$pricingGroup->getDomainId()][$pricingGroup->getId()] = new ProductSellingPrice(
+            $productSellingPrices[$pricingGroup->getDomainId()] =
+                $this->getAllProductSellingPricesByDomainId($product, $pricingGroup->getDomainId());
+        }
+
+        return $productSellingPrices;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param int $domainId
+     * @return \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductSellingPrice[]
+     */
+    public function getAllProductSellingPricesByDomainId(Product $product, $domainId)
```

Comment:
Please use type-hinting in the new code whenever possible. See [Code Quality Principles](https://github.com/shopsys/shopsys/blob/v7.2.0/docs/contributing/code-quality-principles.md#strict-types) in our docs for details.

```suggestion
    public function getAllProductSellingPricesByDomainId(Product $product, int $domainId): array
```

Reply:
Ok, done